cirq/circuits/frozen_circuit.py

```diff
@@ -85,13 +89,19 @@ def moments(self) -> Sequence['cirq.Moment']:
     def device(self) -> devices.Device:
         return self._device
 
+    @property
+    def name(self) -> Optional[str]:
+        return self._name
+
     def __hash__(self):
-        return hash((self.moments, self.device))
+        return hash((self.moments, self.device, self.name))
+
+    def _serialization_name_(self):
+        return self._name or 'Circuit'
 
-    def serialization_key(self):
-        # TODO: use this key in serialization and support user-specified keys.
+    def diagram_name(self):
```

Comment:
Is this method still intended to be used for serialization? If so, consider a name that indicates this or add a new method that forwards functionality here. Alternatively, maybe a name that declares what this value is rather than how it's used?

Reply:
This method is only used in circuit diagrams now - hence the name. Serialization instead uses `_serialization_name_`.